batch_processing.sh

```diff
@@ -0,0 +1,172 @@
+#!/bin/bash
+# NB: This file is best displayed with 120 col.
+# 
+# Processing of dataset of MS patients. Data organized with BIDS. This script is designed to be run 
+# across multiple subjects in parallel using 'sct_run_batch', but it can also be used to run processing on a single 
+# subject. The input data is assumed to be in BIDS format.
+# 
+# IMPORTANT: This script MUST be run from the root folder of the repository, because it relies on Python scripts located 
+#  in the root folder.
+#
+# Usage:
+#   ./batch_processing.sh <SUBJECT>
+#
+# Example:
+#   ./batch_processing.sh sub-03
+#
+# Author: Julien Cohen-Adad
+
+# Parameters
+# TODO: remove following parameter if not needed
+# vertebral_levels="1:3"  # Vertebral levels to extract metrics from. "2:12" means from C2 to T5 (included)
+# List of tracts to extract:
+# TODO: deal with that later
+tracts=(
+  "32,33"\
+  "51"\
+  "52"\
+  "53"\
+  "4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29"\
+  "30,31"\
+  "34,35"\
+  "4,5"\
+  "4,5,8,9,10,11,16,17,18,19,20,21,22,23,24,25,26,27"\
+  "0,1,2,3,6,7,12,13,14,15"\
+)
+# The following global variables are retrieved from the caller sct_run_batch but could be overwritten by 
+# uncommenting the lines below:
+# PATH_DATA_PROCESSED="~/data_processed"
+# PATH_RESULTS="~/results"
+# PATH_LOG="~/log"
+# PATH_QC="~/qc"
+
+# Uncomment for full verbose
+# set -v
+
+# Immediately exit if error
+set -e
+
+# Exit if user presses CTRL+C (Linux) or CMD+C (OSX)
+trap "echo Caught Keyboard Interrupt within script. Exiting now.; exit" INT
+
+# Save script path
+PATH_SCRIPT=$PWD
+
+
+# CONVENIENCE FUNCTIONS
+# =====================================================================================================================
+
+label_vertebrae_if_does_not_exist() {
+  # This function checks if a manual label file already exists, then:
+  #   - If it does, copy it locally.
+  #   - If it doesn't, perform automatic labeling.
+  # This allows you to add manual labels on a subject-by-subject basis without disrupting the pipeline.
+
+  local file="${1}"
+  local file_seg="${2}"
+  # Update global variable with segmentation file name
+  FILELABEL="${file}"_label-disc
+  FILELABELMANUAL="${PATH_DATA}"/derivatives/labels/"${SUBJECT}"/anat/"${FILELABEL}".nii.gz
```

Comment:
I am wondering if we might have some problems here in case they have a session folder? It might be the case with their data so I am not modifying the code for now.